### DISCUSSION TITLE
How to specify the voltage between electrodes/source frequency of ERT simulation or the source frequency/waveform in TT simulation?

### DISCUSSION THREAD

**Discussion by jinshanmu** [gimli-org/pyGIMLi#646] [Q&A]
I'm going through the pygimli tutorials/examples and cannot find parameters to set these values in functions like ert.createData() and ert.simulate() or tt.createRAData() and tt.simulate(). How should I do it?

A new pygimli user here and thank you in advance!

**Comment by halbmy** [comment]
For the numerical solution of the DC resistivity (ERT) problem, both the voltage and the frequency are not needed as one simulates resistances and considers the resistivity attributed to the measuring frequency. The voltage gain becomes important when estimating errors of field data but not in simulations. The frequency can only become interesting if induced polarization effects are taken into account and several frequencies are used.

**Reply by jinshanmu** [reply]
I'm still a bit confused about the "... and considers the resistivity attributed to the measuring frequency" part, as frequency is not needed in the simulation. Can you further explain on it?🙏🏻

**Reply by halbmy** [reply]
it solves the DC (direct current, i.e. frequency zero) resistivity problem. Actually one needs to have a frequency > 0 to which of course the resulting resistivity is associated.

**Reply by jinshanmu** [reply]
Got it. Thanks!

**Comment by halbmy** [comment]
As to the traveltime problem: the traveltime manager does not simulate the full waveform, but only the first arrivals, i.e. the solution of the Eikonal equation which does not take the frequency content and waveform into account. The frequency can be accounted for by so-called fat-rays. See notebooks on https://github.com/gimli-org/notebooks/tree/main/traveltime
